### PR TITLE
docs: Proposal to add installation steps for local development using Kind

### PR DESCRIPTION
**Pull request by revitalbarletz**
Adding some steps for running Argo CD locally using Kind.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 ## 1. Install Argo CD
 
+### Normal Install
```
Comment (revitalbarletz) [inline docs/getting_started.md:14]: Will be happy to replace 'Normal' with some other term... any suggestion is greatly appreciated!

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
@revitalbarletz This is a great addition! I think it needs to be in a separate page now that I see how long it is. What do you think about creating a "Try Argo CD Locally" page and then linking to it from this getting started page?

File: docs/getting_started.md (hunk 1 above)
Comment (kostis-codefresh) [reply docs/getting_started.md:14]: I propose either "Default" or "Local" install.

**Comment by revitalbarletz** [comment]
@todaywasawesome thanks for the feedback, done.

File: docs/try_argo_cd_locally.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Try Argo CD Locally
+
+!!! tip
+    This guide assumes you have a grounding in the tools that Argo CD is based on. Please read [understanding the basics](understand_the_basics.md) to learn about these tools.
+
+
+Follow these steps to install `Kind` for local development and set it up with Argo CD.
```
Comment (todaywasawesome) [inline docs/try_argo_cd_locally.md:7]:
```suggestion
!!! tip
    This guide assumes you have a grounding in the tools that Argo CD is based on. Please read [understanding the basics](understand_the_basics.md) to learn about these tools.


Follow these steps to install `Kind` for local development and set it up with Argo CD.

To run an Argo CD development environment [review the developer guide for running locally](../developer-guide/running-locally).
```

**Review by todaywasawesome — APPROVED** [review]

File: docs/try_argo_cd_locally.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+To run an Argo CD development environment [review the developer guide for running locally](../developer-guide/running-locally).
+
+## Install Kind
+
+```bash
+curl -Lo ./kind https://kind.sigs.k8s.io/dl/latest/kind-darwin-amd64
```
Comment (rumstead) [inline docs/try_argo_cd_locally.md:14]: Should we link to installing kind here? We are assuming everyone is running a mac. 

https://kind.sigs.k8s.io/docs/user/quick-start#installation

**Review by rumstead — APPROVED** [review]
LGTM!

**Review by andrii-korotkov-verkada — APPROVED** [review]

**Review by todaywasawesome — APPROVED** [review]

**Comment by revitalbarletz** [comment]
Thank you @todaywasawesome ! This is my first PR that got merged !!!

**Pull request merged**